File: src/main/java/me/Geek/api/mail/Mail.java
package me.Geek.api.mail;


import org.bukkit.inventory.ItemStack;

import java.util.ArrayList;
import java.util.List;
import java.util.UUID;

/**
 * 作者: 老廖
 * 时间: 2022/7/28
 **/
public interface Mail extends Cloneable {
    UUID getMailID();

    MailType getMailType();

    String getTitle();

    UUID getSender();

    UUID getTarget();

    String getText();

    String getState();

    void setState(String state);

    void setTarget(UUID target);
    void setMailID(UUID mailID);

    String getAppendix();
    String getType();

    default double getMoney() {
        return 0;
    }

    default int getPoints() {
        return 0;
    }

    default int getExp() {
        return 0;
    }

    default ItemStack[] getItemStacks() {
        return new ItemStack[0];
    }

    default List<String> getCommand() {
        return new ArrayList<>();
    }

    void SendMail();

    void giveAppendix();



}

File: src/main/kotlin/me/geek/mail/api/mail/sub/Mail.kt
package me.geek.mail.api.mail.sub

import org.bukkit.entity.Player
import org.bukkit.inventory.ItemStack
import java.util.*

/**
 * 作者: 老廖
 * 时间: 2022/7/28
 */
interface Mail : Cloneable {
    val mailID: UUID
    val mailType: String

    val title: String
    val text: String
    val sender: UUID
    val target: UUID
    var state: String
    val senderTime: String
    var getTime: String
    val appendixInfo: String

    val permission: String
      get() = "mail.global"

    val additional: String?
      get() = " "
    val itemStacks: Array<ItemStack>?
      get() = null
    val command: List<String>?
      get() = null



    fun sendMail()
    fun giveAppendix()
    fun condition(player: Player, appendix: String): Boolean

}